### PR TITLE
fix ambiguous global enum definition for SPI in OSPIFBlockDevice driver

### PR DESCRIPTION
**Pull request by timIdeaTech**
### Summary of changes 

OSPIFBlockDevice.h defined "SPI" as a global enum under the name _mode. 

Both are collision/ambiguity hazards so, using the global enum convention of the existing driver code
```
enum _mode {
    SPI,
    SOPI,
    DOPI
};

```
has been changed to:
```
enum ospif_opi_mode {
    OSPIF_OPI_MODE_SPI = 0, /* SPI mode (OPI modes disabled)*/
    OSPIF_OPI_MODE_SOPI,    /* STR-OPI mode */
    OSPIF_OPI_MODE_DOPI     /* DTR-OPI mode */
};

```
as well as any references in OSPIFBlockDevice.cpp

#### Impact of changes 

Any reference to the previous enum's in user code will result in preprocessor not defined errors.

#### Migration actions required 

All references to the previous enum's in user code will have to be updated with the new prefix OSPIF_OPI_MODE_

### Documentation 

Update references in change_mode() references in https://mbed-ce.github.io/mbed-os/class_o_s_p_i_f_block_device.html)

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 

@multiplemonomials 
----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by timIdeaTech** [comment]
Not sure I follow the 2 failed checks above. I emulated the other enum comments for style and I'm not sure what the other is...

If I can fix it, let me know. If not, let me know how I can avoid this in the future.

I'm assuming I'm not on the hook for the mbed-ce html doc changes, correct?

File: tools/doxygen-awesome-css
Comment (multiplemonomials) [inline tools/doxygen-awesome-css:1]: Wait, why change the doxygen-awesome-css hash?

**Comment by multiplemonomials** [comment]
For the style check it appears to be complaining about the space after the asterisk.  Dang astyle it can be so finicky sometimes...

**Comment by multiplemonomials** [comment]
As for the other error, it would appear that on [this line](https://github.com/mbed-ce/mbed-os/blob/65519018b0cf1818e0ff10aa928936bc3485d967/storage/blockdevice/COMPONENT_SD/include/SD/SDBlockDevice.h#L115), `#SPI` needs to be changed to `#mbed::SPI`.  I think that Doxygen was linking it to this enum instead of the SPI class!

**Comment by timIdeaTech** [comment]
Sorry, just figuring this all out. Should be good to merge the 3 files except doxygen-awesome-css. Not sure how to back out that one.

File: tools/doxygen-awesome-css
Comment (multiplemonomials) [reply tools/doxygen-awesome-css:1]: Hmm I think what happened is you may have done something like
```
git submodule foreach git pull origin master
```
perhaps?  Anyway, it appears that this submodule got updated to point at the latest version of this repo -- which is not in and of itself a bad thing, but doesn't belong in this PR.

Luckily, fixing it is pretty easy,  We just need to update the submodule pointer to point at the original commit, then commit and push the change:
```
$ cd mbed-os/tools/doxygen-awesome-css
$ git reset --hard a3c119b4 
$ cd ../..
$ git add tools/doxygen-awesome-css
$ git commit -m "Fix submodule hash"
$ git push
```

That should do the trick, and then I'm good to merge!

**Review by multiplemonomials — APPROVED** [review]

**Pull request merged**